presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -1796,6 +1796,28 @@ public AnalyzePropertyManager getAnalyzePropertyManager()
         return analyzePropertyManager;
     }
 
+    @Override
+    public Optional<String> redirectCatalog(Session session, QualifiedObjectName tableName)
+    {
+        requireNonNull(tableName, "tableName is null");
+
+        if (tableName.getCatalogName().isEmpty() || tableName.getSchemaName().isEmpty() || tableName.getObjectName().isEmpty()) {
+            return Optional.empty();
+        }
+
+        Optional<CatalogMetadata> catalog = getOptionalCatalogMetadata(session, tableName.getCatalogName());
```

Comment:
`metadata.redirectCatalog(connectorSession, tableName.asSchemaTableName())` returns `Optional<String>` instead of `String`, making it difficult to follow the pattern here.